src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -808,6 +809,35 @@ public Solution WithProjectCompilationOptions(ProjectId projectId, CompilationOp
 
         /// <summary>
         /// Create a new solution instance with the project specified updated to have
+        /// the specified emit options.
+        /// </summary>
+        public Solution WithProjectEmitOptions(ProjectId projectId, EmitOptions options)
+        {
+            if (projectId == null)
+            {
+                throw new ArgumentNullException(nameof(projectId));
+            }
+
+            if (options == null)
```

Comment:
Is null bad?  I thought previously you wrote that that meant you got the default emit optins.